google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/ConnectionImpl.java

```diff
@@ -1885,7 +1940,11 @@ private void popUnitOfWorkFromTransactionStack() {
   }
 
   private ApiFuture<Void> executeDdlAsync(CallType callType, ParsedStatement ddl) {
-    return getOrStartDdlUnitOfWork().executeDdlAsync(callType, ddl);
+    ApiFuture<Void> result = getOrStartDdlUnitOfWork().executeDdlAsync(callType, ddl);
+    // reset proto descriptors after executing a DDL statement
+    this.protoDescriptors = null;
+    this.protoDescriptorsFilePath = null;
```

Comment:
Would it make more sense to move this to `setDefaultTransactionOptions()`? It keeps the code cleaner, as we have all the reset code in one method, but it does mean that executing for example a query after setting the proto descriptors also clears the fields. One option to prevent the latter could be to add an input argument to `setDefaultTransactionOptions()` that indicates the type of statement that was executed, and only reset these fields if the last statement was a DDL statement.

Reply:
sure, will check if it makes sense to move that logic to `setDefaultTransactionOptions()` as a seperate PR. This PR is very big that the code review will be difficult. Will work on this immediately.

Reply:
This comment will be addressed in PR https://github.com/googleapis/java-spanner/pull/3126